### ISSUE TITLE
Minimum supported browser versions?

### ISSUE THREAD

**Issue by christianp** [jsxgraph/jsxgraph#644]
Is there a list of minimum supported browser versions, or of features that JSXGraph uses?

We had a report from a user of the Numbas JSXGraph extension that students with running old versions of MacOS encountered an error because the `Array.at` method, used in [some code in the Text element](https://github.com/jsxgraph/jsxgraph/blob/main/src/base/text.js#L324), wasn't defined. This particular code could easily be replaced by indexing with square brackets instead, but I'd rather be able to say to the users what minimum versions are supported.

For Numbas, I made a tool which lets me pick the browser features I'm using, and shows the minimum versions of each browser that support them: https://somethingorotherwhatever.com/can-i-also-use/
I aim for compatibility with 95% of devices, weighted by frequency of use.
The settings file is checked into the Numbas repository, to document the required features: https://github.com/numbas/Numbas/blob/master/can-i-also-use-settings.json.
Would you be interested in doing this?

**Comment by alfredwassermann** [comment]
Dear Christian,

indeed, `String.at` had been introduced used in JSXGraph recently. But in this week it was kicked out again, because Moodle < 4.4 came with its own `String.at` implementation which did not support negative values of the parameter. Beside this, we have 2 variables  declared with `let` in the new `mono_thin` theme (since v1.8.0). We hope to get rid of them, too.
Apart from these, JSXGraph support should go back to least IE 9 (if we did not miss something). 
That is, the upcoming v1.8.1 will be backwards compatible again.

I would be happy to test your site.

Best wishes,
Alfred

**Comment by christianp** [comment]
Great, thanks!

**Comment by christianp** [comment]
We've had a report of a Numbas question using JSXGraph not working in Mi Browser 12.13.2. Other Numbas questions without JSXGraph work for them. We're using JSXGraph 1.7.0.

I can't find any compatibility information for Mi browser, but looking at the user agent string, it seems to be based on Chrome 89. 

Do you have a way of testing with old versions of Chrome?

**Comment by alfredwassermann** [comment]
Indeed, I could create an error with JSXGraph v1.7.0 in Chrome 89. Unfortunately, this browser version also does not like the method `String.at()`. Thus, this might be the problem with the Mi browser, too.

I could test the browser by installing ubuntu 20.4 in virtualbox and install chrome 89 from https://mirror.cs.uchicago.edu/google-chrome/pool/main/g/google-chrome-stable/.

**Comment by christianp** [comment]
OK, thanks. So I should just upgrade JSXGraph, I think.

**Comment by alfredwassermann** [comment]
If you send me a public link, I could test the page if there are further problems.
